sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BaseJoinExec.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.joins
+
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.catalyst.plans.JoinType
+import org.apache.spark.sql.execution.{BinaryExecNode, ExplainUtils}
+
+/**
+ * Holds common logic for join operators
+ */
+trait BaseJoinExec extends BinaryExecNode {
+  def joinType: JoinType
+  def condition: Option[Expression]
+
+  override def simpleStringWithNodeId(): String = {
+    val opId = ExplainUtils.getOpId(this)
+    s"$nodeName $joinType ($opId)".trim
+  }
+
+  override def verboseStringWithOperatorId(): String = {
+    val joinCondStr = if (condition.isDefined) {
+      s"${condition.get}"
+    } else "None"
+    s"""
+       |(${ExplainUtils.getOpId(this)}) $nodeName ${ExplainUtils.getCodegenId(this)}
+       |Join condition : ${joinCondStr}
+     """.stripMargin
+  }
```

Comment:
We need this trait for join physical plans? There seems to be less common method between them...

Reply:
Yea, when implementing the suggested follow-up, I found only `joinType` and `conditions` can be pull up to base trait. And even the `joinType` not found in `CartesianProductExec`. 

@maryannxue Does this PR meets your suggestion? Just let me know if I misunderstood anything, thanks! :-)

Reply:
maybe we can also put left/right join keys here. The join keys are Nil in some sub-classes like `CartesianProductExec`.

Reply:
Thank you @Eric5553! this looks good. Adding equi-join keys sounds good to me too.

Reply:
@cloud-fan @maryannxue left/right join keys are also added to `BaseJoinExec`